SignalMessaging/environment/VersionMigrations.m

```diff
@@ -204,6 +210,24 @@ + (void)updatePublicChatMapping
     }];
 }
 
+# pragma mark Loki - Upgrading Friend Request Status
+
+
+// Versions less than or equal to 1.1.1 stored friend request status on the thread
++ (void)moveFriendRequestStatusFromThread
+{
+    OWSPrimaryStorage *storage = OWSPrimaryStorage.sharedManager;
+    [OWSPrimaryStorage.dbReadWriteConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction * _Nonnull transaction) {
```

Comment:
This is a very sensitive operation. If it goes wrong it could mess up existing users' contacts. We'll need to definitely do an internal test before shipping this.

Also, where exactly will this get invoked from? Do we know there won't already be an ongoing write transaction?

Reply:
Fixed this. Was doing the migration in the wrong place.